src/lib/x509/x509path.h

```diff
@@ -50,13 +50,16 @@ class BOTAN_PUBLIC_API(2, 0) Path_Validation_Restrictions final {
       *        If zero, there is no maximum age
       * @param trusted_ocsp_responders certificate store containing certificates
       *        of trusted OCSP responders (additionally to the CA's responders)
+      * @param ignore_trusted_root_time_range if true, validity checks on the
+      *        time range of the trusted root certificate only produce warnings
       */
       Path_Validation_Restrictions(
          bool require_rev = false,
          size_t minimum_key_strength = 110,
          bool ocsp_all_intermediates = false,
          std::chrono::seconds max_ocsp_age = std::chrono::seconds::zero(),
-         std::unique_ptr<Certificate_Store> trusted_ocsp_responders = std::make_unique<Certificate_Store_In_Memory>());
+         std::unique_ptr<Certificate_Store> trusted_ocsp_responders = std::make_unique<Certificate_Store_In_Memory>(),
+         bool ignore_trusted_root_time_range = false);
```

Comment:
Unrelated to this PR, this constructor is getting out of hand - we might want to consider introducing a builder-style API for this.

Reply:
I've created a ticket (#3939) to collect API improvement ideas for X.509.